src/strings/tests-nfa-noodlification.cc

```diff
@@ -449,6 +449,28 @@ TEST_CASE("Mata::Nfa::SegNfa::noodlify_for_equation() both sides") {
             }
         }
     }
+
+    SECTION("Simple automata -- regex 1") {
+        Nfa x, y, z, u;
+        create_nfa(&x, "a");
+        create_nfa(&y, "ab*");
+        create_nfa(&z, "ab*");
+        create_nfa(&u, "a*");
+
+        auto res = std::vector<std::vector<std::pair<Nfa, SegNfa::EpsCntVector>>>( {
+                {{x, {0, 0} }, {x, {1, 1} }},
+            } );
+        SegNfa::NoodleSubstSequence noodles = SegNfa::noodlify_for_equation(
+            std::vector<std::shared_ptr<Nfa>>{std::make_shared<Nfa>(x), std::make_shared<Nfa>(y) }, 
+            std::vector<std::shared_ptr<Nfa>>{std::make_shared<Nfa>(z), std::make_shared<Nfa>(u)});
+        CHECK(noodles.size() == 1);
+        for(size_t i = 0; i < noodles.size(); i++) {
+            for(size_t j = 0; j < noodles[i].size(); j++) {
+                CHECK(noodles[i][j].second == res[i][j].second);
+                CHECK(are_equivalent(*noodles[i][j].first.get(), res[i][j].first, nullptr));
```

Comment:
You do not need to do `.get()` here, shared pointers act as pointers, so `*noodles[i][j].first` is enough.

Reply:
You are right. It is a syntactic sugar.

@vhavlena Do you want to address it, or will you only take into consideration and keep it as it is? If you choose the latter, I believe we can merge the PR.

Reply:
Yes, comparison of shared pointers is done via comparison of .get() addresses. Let's merge it.